src/Query.tsx

```diff
@@ -349,14 +349,8 @@ export default class Query<TData = any, TVariables = OperationVariables> extends
         this.updateCurrentData();
       },
       error: error => {
-        if (!this.lastResult) {
-          // We only want to remove the old subscription, and start a new
-          // subscription, when an error was received and we don't have a
-          // previous result stored. This means either no previous result was
-          // received due to problems fetching data, or the previous result
-          // has been forcefully cleared out.
-          this.resubscribeToQuery();
-        }
+        this.resubscribeToQuery();
```

Comment:
I haven't added any additional tests to validate this behavior, as switching to always resubscribe in the error callback here lead to existing test failures. These were then fixed by clearing query store errors below.